connector_acp/models/etd_mixin.py

```diff
@@ -92,19 +90,19 @@ def build_xml(self):
             'datas_fname': xml_file.filename,
             'res_model': self._name,
             'res_id': self.id})
-        # TODO: Uncomment once the XSD files are properly formed
-        # Check the rendered XML against the XSD
-        # xsd = base64.b64decode(document_id.xsd).decode('utf-8')
-        # try:
-        #     xmlschema = etree.XMLSchema(xsd)
-        #     xml_doc = etree.fromstring(xml)
-        #     result = xmlschema.validate(xml_doc)
-        #     if not result:
-        #         xmlschema.assert_(xml_doc)
-        #     return xml
-        # except AssertionError as e:
-        #     _logger.warning(etree.tostring(xml_doc))
-        #     raise UserError(_("XML Malformed Error: %s") % e.args)
+        if etd_id.xsd:
+            # Check the rendered XML against the XSD
+            xsd = base64.b64decode(etd_id.xsd).decode('utf-8')
+            try:
+                xmlschema = etree.XMLSchema(xsd)
+                xml_doc = etree.fromstring(xml)
+                result = xmlschema.validate(xml_doc)
+                if not result:
+                    xmlschema.assert_(xml_doc)
+                return xml
```

Comment:
Revisiting this, we have a bug: the return statement should be after the `if std_id.xsd` block.
Otherwise if no xsd, nothing is returned.

Reply:
@dreispt Should be fixed and support the definition of multiple files. The generation is not there yet.